Metrist-Software/product-docs
# Harden manifests to markdown renderer

## Description

This PR accomplishes a few things:

1. The earlier prototype of `manifests-to-markdown` is removed.
2. It is replaced by a more streamlined feature (without file watching or hot-reloading of monitors docs);
4. Adds newly rendered monitor docs to VitePress from https://assets.metrist.io/dist/monitors/manifests-preview.json.

Note: /manifests.json and /manifests-preview.json are currently different at assets.metrist.io/dist/monitors. This is worth studying, perhaps I'm just observing a caching issue. /manifests-preview.json is more complete at the moment.

## Jira Items

PDD 6-63, 65-67

## Tested (If not tested in dev, explain)

[X] local

## Testing

To confirm everything works, I suggest:

1. Pull branch to local
2. In root folder:
  - `npm install` (**to get new dependencies**)
  - `npm run dev` : Note ~80 files are rendered (**this may take a few seconds**)
  - (make it more interesting)
  - delete all `metrist_*.md` files in vitepress/docs/monitors (just don't delete the index.md in that folder)
  - run `npm run dev` and confirm the files are re-created, and you can view them in a browser: http://localhost:5173/monitors/metrist_gke.html etc.
  - also run `npm run coverage`, `npm run build`.

## Deployment steps

1. Merge & delete branch
